client/client_session.js

```diff
@@ -0,0 +1,272 @@
+var log = require('minilog')('radar:client')
+var EventEmitter = require('events').EventEmitter
+var inherits = require('util').inherits
+var ClientSessionStateMachine = require('./client_session_state_machine')
+
+function ClientSession (name, id, accountName, version, transport) {
+  this.state = ClientSessionStateMachine.create(this)
+
+  this.createdAt = Date.now()
+  this.lastModified = Date.now()
+  this.name = name
+  this.accountName = accountName
+  this.id = id
+  this.subscriptions = {}
+  this.presences = {}
+  this.version = version || '0.0.0'
+  EventEmitter.call(this)
+
+  this.transport = transport
+  this._setupTransport()
+  clientsById[this.id] = this
+}
+
+inherits(ClientSession, EventEmitter)
+
+ClientSession.prototype._initialize = function (set) {
+  var self = this
+
+  if (set) {
+    Object.keys(set).forEach(function (key) {
+      self[key] = set[key]
+    })
+  }
+
+  if (this.state.can('initialize')) {
+    this.state.initialize()
+  }
+  this.lastModified = Date.now()
+}
+
+ClientSession.prototype._cleanup = function () {
+  delete clientsById[this.id]
```

Comment:
I removed the name associations, see https://github.com/zendesk/radar/pull/198#issuecomment-168849148